src/components/DomainSearchResultsDisplay.tsx

```diff
@@ -46,3 +47,27 @@ export const DomainSearchResultsDisplay: React.FC<DomainSearchResultProps> = ({
     </div>
   );
 };
+
+function mergeRecordsByValue(records: RecordWithSelector[]): RecordWithSelector[] {
+  const valueMap = new Map<string, RecordWithSelector>();
+
+  records.forEach((record) => {
+    if (valueMap.has(record.value)) {
```

Comment:
You can avoid a double map lookup + null assertion by writing like this instead:

```
    const existing = valueMap.get(record.value);
    if (existing) {
```